apps/src/applab/setPropertyDropdown.js

```diff
@@ -233,7 +272,16 @@ for (var elementType in PROPERTIES) {
     }
     elementProperties.infoForFriendlyName[friendlyName] = PROP_INFO[propName];
     if (!PROP_INFO[propName].alias) {
-      elementProperties.dropdownOptions.push('"' + friendlyName + '"');
+      let setValueParam = DEFAULT_PROP_VALUES[friendlyName];
+      if (setValueParam) {
+        elementProperties.dropdownOptions.push({
+          text: '"' + friendlyName + '"',
+          display: '"' + friendlyName + '"',
+          setValueParam: setValueParam,
+        });
+      } else {
+        elementProperties.dropdownOptions.push('"' + friendlyName + '"');
```

Comment:
There's some _slightly_ complex logic here around generating a string or an object, and it seems to be roughly duplicated here and where we generate `fullDropdownOptions` above.  Can we deduplicate this somehow?